lgc/util/StartStopTimer.cpp

```diff
@@ -42,13 +43,17 @@ namespace {
 
 // =====================================================================================================================
 // Pass to start or stop a timer
-class StartStopTimer : public ModulePass {
+class StartStopTimer : public PassInfoMixin<StartStopTimer> {
 public:
-  static char ID;
-  StartStopTimer() : ModulePass(ID) {}
-  StartStopTimer(Timer *timer, bool starting) : ModulePass(ID), m_timer(timer), m_starting(starting) {}
+  StartStopTimer() {}
+  StartStopTimer(Timer *timer, bool starting) : m_timer(timer), m_starting(starting) {}
 
-  bool runOnModule(Module &module) override;
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &AM) {
+    runOnModule(M);
+    return PreservedAnalyses::all();
+  }
```

Comment:
Defining this method in the class implicitly declares it as `inline` which is required to avoid the `unused-function` warning.

Reply:
I am not sure what you mean. I think you need to declare the method here only and define it later outside of the class.

Reply:
The problem is that this function is not used yet in the codebase because LLVM's new pass manager is not used yet. Because the function is not used and defined in a cpp file, we get an `unused function` warning that causes an error with `-Wall`. Defining it directly inside the class implicitly declares it as `inline`, and the warning is not emitted for inline functions. I think we'll eventually need to move that class definition to its own .h file anyway but we can do that later when we switch to the new PM and actually use that function.

Reply:
Ah, now I see what you meant. Thanks.